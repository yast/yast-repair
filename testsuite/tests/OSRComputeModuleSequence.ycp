/*
 * YaST2: Modules testsuite
 *
 * Description:
 *   Testsuite for YaST2 OS Repair: OSRComputeModuleSequence() in file osr_system.ycp
 *
 * Authors:
 *   Michael Koehrmann <curry@suse.de>
 */
{
    include "testsuite.ycp";
    include "osr/osr_system.ycp";

    map  osr_map    = $[];
    map  provides_1 = $[];
    map  provides_2 = $[];
    map  provides_3 = $[];
    list requires_1 = [];
    list requires_2 = [];
    list requires_3 = [];

    // Test 1
    provides_1 = $["a" : $[], "b" : $[]];
    requires_1 = [];
    provides_2 = $["c" : $[], "d" : $[]];
    requires_2 = ["a"];
    provides_3 = $["e" : $[]];
    requires_3 = ["b", "c"];

    osr_map = $[
		    "module_list" : [ "module_1", "module_2", "module_3" ],
		    "module_sequence" : [],
		    "module_1" : $[
				   "requires" : requires_1,
				   "provides" : provides_1,
		    ],
		    "module_2" : $[
				   "requires" : requires_2,
				   "provides" : provides_2,
		    ],
		    "module_3" : $[
				   "requires" : requires_3,
				   "provides" : provides_3,
		    ]
    ];
    
    TEST(``(OSRComputeModuleSequence()), [], nil);
    TEST(``(OSRTestGetOsrMap()), [], nil);

    // Test 2
    provides_1 = $["a" : $[], "b" : $[]];
    requires_1 = [];
    provides_2 = $["c" : $[], "d" : $[]];
    requires_2 = ["a", "e"];
    provides_3 = $["e" : $[]];
    requires_3 = ["b", "c"];

    osr_map = $[
		    "module_list" : [ "module_1", "module_2", "module_3" ],
		    "module_sequence" : [],
		    "module_1" : $[
				   "requires" : requires_1,
				   "provides" : provides_1,
		    ],
		    "module_2" : $[
				   "requires" : requires_2,
				   "provides" : provides_2,
		    ],
		    "module_3" : $[
				   "requires" : requires_3,
				   "provides" : provides_3,
		    ]
    ];
    
    TEST(``(OSRComputeModuleSequence()), [], nil);
    TEST(``(OSRTestGetOsrMap()), [], nil);
    
}
