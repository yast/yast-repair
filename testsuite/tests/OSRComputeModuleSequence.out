Return	true
Return	$["global_provides_list":["a", "b", "c", "d", "e"], "module_1":$["provides":$["a":$[], "b":$[]], "requires":[]], "module_2":$["provides":$["c":$[], "d":$[]], "requires":["a"]], "module_3":$["provides":$["e":$[]], "requires":["b", "c"]], "module_list":[], "module_sequence":["module_1", "module_2", "module_3"]]
Log	ERROR OSRComputeModuleSequence: module(s) not added to the sequence list: ["module_2", "module_3"]
Return	false
Return	$["global_provides_list":["a", "b"], "module_1":$["provides":$["a":$[], "b":$[]], "requires":[]], "module_2":$["provides":$["c":$[], "d":$[]], "requires":["a", "e"]], "module_3":$["provides":$["e":$[]], "requires":["b", "c"]], "module_list":["module_2", "module_3"], "module_sequence":["module_1"]]
